api/V1/Device.js

```diff
@@ -239,4 +239,67 @@ module.exports = function(app, baseUrl) {
       }
     })
   );
+
+  /**
+   * @api {post} /api/v1/devices/rename Change the name and group of a device.
+   * @apiName RenameDevice
+   * @apiGroup Device
+   * @apiDescription This call can change the name and the group of a device.
+   *
+   * @apiUse V1DeviceAuthorizationHeader
+   *
+   * @apiParam {String} newName new name of the device.
+   * @apiParam {String} newGroup name of the group you want to move the device to.
+   *
+   * @apiUse V1ResponseSuccess
+   * @apiUse V1ResponseError
+   */
+  app.post(
+    apiUrl + "/rename",
+    [
+      auth.authenticateDevice,
+      middleware.getGroupByName(body, "newGroup"),
+      middleware.checkNewName("newName"),
+    ],
+    middleware.requestWrapper(async function(request, response) {
+
+      const newName = request.body.newName;
+      const newGroup = request.body.group;
+
+      const conflictingDevice = await models.Device.findOne({
+        where: {
+          devicename: newName,
+          GroupId: newGroup.id,
+        }
+      });
+
+      const successResponse = {
+        statusCode: 200,
+        devicename: request.body.newName,
+        groupname: newGroup.groupname,
+        messages: ["name and gruop set"],
+      };
+
+      if (conflictingDevice != null) {
+        if (conflictingDevice.id == request.device.id) {
+          return responseUtil.send(response, successResponse);
+        } else {
+          return responseUtil.send(response, {
+            statusCode: 400,
+            messages: ["already a deice in group '"+newGroup.groupname+"' with the name '"+newName+"'"]
+          });
+        }
+      }
+
+      if (await request.device.rename(newName, newGroup.id)) {
```

Comment:
Can you move the check for conflicts to `Device.rename` please? It could throw a ClientError error if the name is already taken. 

It would also be more robust if you could do the check for a conflict into a SERIALIZABLE transaction with the update. That way there's no chance of concurrent renames or other updates messing things up.

- https://sequelize.org/master/manual/transactions.html#managed-transaction--auto-callback-
- https://sequelize.org/master/manual/transactions.html#isolation-levels